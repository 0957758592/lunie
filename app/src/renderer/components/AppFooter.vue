<template>
  <div class="app-footer">
    <div class="status-sync">
      <div v-if="statusSynced"><i class="fa fa-spin fa-refresh"></i> Syncing&hellip;</div>
      <div v-else>Your balance: 0.00 (TODO)</div>
    </div>
    <div class="menu">
      <a href="#"><i class="fa fa-lock"></i></a>
      <router-link to="/settings"><i class="fa fa-gear"></i></router-link>
      <a href="#"><i class="fa fa-user-secret"></i></a>
      <div class="status-online" :class="{ online: statusOnline }">
        <i v-if="statusOnline" class="fa fa-link"></i>
        <i v-else="statusOnline" class="fa fa-chain-broken"></i>
      </div>
    </div>
  </div>
</template>

<script>
import Btn from '@nylira/vue-button'
export default {
  components: {
    Btn
  },
  data () {
    return {
      statusSynced: true,
      statusOnline: true
    }
  },
  methods: {
    go (route) {
      this.$router.push(route)
    }
  }
}
</script>

<style lang="stylus">
@require '../styles/variables.styl'

.app-footer
  height 2rem + 0.0625rem
  border-top 1px solid bc
  background c-app-fg
  display flex
  align-items center

  padding 0 0.5rem

  .status-sync
    flex 1
    font-size 0.875rem

  .menu
    display flex

    a, .status-online
      color dim
      height 2rem
      width 2rem
      display flex
      align-items center
      justify-content center

    a
      &:hover
        color link

    .status-online
      i.fa
        color hsl(0,100%,35%)
      &.online i.fa
        color hsl(120,100%,35%)
</style>
