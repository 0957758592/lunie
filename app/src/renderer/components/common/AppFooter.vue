<template lang='pug'>
footer.app-footer
  .container
    .copyright &copy; 2017 Interchain Foundation
</template>

<script>
import {mapGetters} from 'vuex'
export default {
  name: 'app-footer',
  computed: {
    ...mapGetters(['syncHeight', 'syncTime', 'syncing', 'numPeers'])
  }
}
</script>

<style lang="stylus">
@require '~variables'

.app-footer
  .container
    border-top px solid bc
    height 3rem
    display flex
    align-items center
    justify-content space-between
    padding 0 1rem

    color dim
</style>
