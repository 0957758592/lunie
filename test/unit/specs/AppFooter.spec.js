import Vuex from 'vuex'
import { mount, createLocalVue } from 'vue-test-utils'
import htmlBeautify from 'html-beautify'
import AppFooter from 'renderer/components/common/AppFooter'
const node = require('renderer/vuex/modules/node').default({
  node: require('../helpers/node_mock')
})

const localVue = createLocalVue()
localVue.use(Vuex)

describe('AppFooter', () => {
  let wrapper, store

  it('has the expected html structure if connected', () => {
    store = new Vuex.Store({
      getters: {
        nodeIP: () => node.state.nodeIP,
        connected: () => node.state.connected,
        lastHeader: () => node.state.lastHeader
      }
    })

    node.state = {
      nodeIP: '192.169.9.5',
      connected: true,
      lastHeader: {
        chain_id: 'gaia-1',
        height: '4555'
      }
    }

    wrapper = mount(AppFooter, {
      localVue,
      store
    })

    expect(htmlBeautify(wrapper.html())).toMatchSnapshot()
  })

  it('has the expected html structure if not connected', () => {
    store = new Vuex.Store({
      getters: {
        nodeIP: () => node.state.nodeIP,
        connected: () => node.state.connected,
        lastHeader: () => node.state.lastHeader
      }
    })

    node.state = {
      nodeIP: '192.169.9.5',
      connected: false,
      lastHeader: {
        chain_id: 'gaia-1',
        height: '4555'
      }
    }

    wrapper = mount(AppFooter, {
      localVue,
      store
    })

    expect(htmlBeautify(wrapper.html())).toMatchSnapshot()
    expect(wrapper.contains('.afi-connecting')).toBe(true)
  })
})
