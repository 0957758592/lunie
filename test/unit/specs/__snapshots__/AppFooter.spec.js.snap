// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFooter has the expected html structure if connected 1`] = `
"<footer class=\\"app-footer\\">
   <div class=\\"app-footer-container\\">
      <div class=\\"afi afi-success\\">
         <i class=\\"material-icons success\\">done</i><span>gaia-1 (#4555)</span>
      </div>
   <div class=\\"afi\\">
      <i class=\\"material-icons\\">settings_ethernet</i><span>192.169.9.5</span>
   </div>
</div>
</footer>"
`;

exports[`AppFooter has the expected html structure if not connected 1`] = `
"<footer class=\\"app-footer\\">
   <div class=\\"app-footer-container\\">
      <div class=\\"afi\\">
         <i class=\\"material-icons fa-spin\\">rotate_right</i><span class=\\"afi-connecting\\">Connectingâ€¦</span>
      </div>
</div>
</footer>"
`;
